File: prism-kotlin-sdk/crypto/src/commonMain/kotlin/io/iohk/atala/prism/kotlin/crypto/HmacSha256.kt
package io.iohk.atala.prism.kotlin.crypto

/**
 * HMAC-SHA-256 facade.
 */
expect object HmacSha256 {

    /**
     * Compute HMAC-SHA-256 data authentication code using shared key.
     */
    fun compute(data: ByteArray, key: ByteArray): ByteArray
}

File: prism-sdk/crypto/shared/src/main/scala/io/iohk/atala/prism/crypto/HmacSha256.scala
package io.iohk.atala.prism.crypto

/**
  * HMAC-SHA-256 facade.
  */
object HmacSha256 {

  /**
    * Compute HMAC-SHA-256 data authentication code using shared key.
    */
  def compute(key: Array[Byte], data: Array[Byte]): Array[Byte] =
    HmacSha256Impl.compute(key, data)
}
